lib/cloudantBaseService.ts

```diff
@@ -115,6 +120,7 @@ export default abstract class CloudantBaseService extends BaseService {
    */
   public setServiceUrl(url: string): void {
     super.setServiceUrl(url);
+    CloudantBaseService.setAgent(this.baseOptions);
```

Comment:
This setting has no point to make here. In the current structure the URL has no influence above the Agent.

Reply:
69652b5